### PR TITLE
Bundle standalone 'manufdb' file to comply with Wireshark changes.

### PR DESCRIPTION
**Pull request by gpotter2**
This PR:
- bundles `manufdb`. This fixes https://github.com/secdev/scapy/issues/4280
- introduces a caching system for files loaded in `data`. This was required in order not to introduce a slow down with this change.
- re-enables all manufdb tests. because we can now always fallback on it.
- `.config/scapy` and `.cache/scapy` now comply with the `XDG_XXX_HOME` variables.

Fun fact: a first version of this PR converted Manuf to a python dictionary. Apparently loading a file containing a very, very big Python dictionnary takes infinetly longer than loading text/binary and parsing it.

### CODE REVIEW

File: scapy/data.py [hunk 1]
```diff
@@ -290,17 +292,69 @@
...
+
+    def _cached_loader(func, name=name):
+        # type: (DecoratorCallable, str) -> DecoratorCallable
+        def load(filename=None):
+            # type: (Optional[str]) -> Any
+            cache_id = hash(filename)
+            if cachepath.exists():
+                try:
+                    with cachepath.open("rb") as fd:
+                        data = pickle.load(fd)
```
Comment (polybassa) [inline scapy/data.py:315]: I think it’s quite dangerous to pickle.load from an arbitrary file. Would it be a possibility to use serialisation to json instead ?

Comment (gpotter2) [reply scapy/data.py:315]: @polybassa @guedou This file is local. If you can edit it you can also add code to `.config/scapy/prestart.py` that is malicious.

Considering this I don't see how it's an issue

Comment (polybassa) [reply scapy/data.py:315]: I totally agree on that point. I just wanted to mention it.

**Review by polybassa — APPROVED** [review]

**Comment by gpotter2** [comment]
Hm, should we `gzip` the blob ? It's a bit big.
I can't recall how supported `gzip` is on all plateforms. If I'm not mistaken `zlib` is often not available...

**Comment by evverx** [comment]
It took me a while but I figured out why Packit failed in https://download.copr.fedorainfracloud.org/results/packit/evverx-scapy-2/fedora-rawhide-x86_64/07331856-scapy/builder-live.log.gz. There is no `~/.config` there. I reproduced it locally by launching a stripped container and running the test there. Once I manually created the `~/.config` directory the tests started passing.

**Comment by gpotter2** [comment]
Thanks for looking into it.

I didn't fully read the XDG spec so I kinda assumed one shouldn't create the `.config`, apparently as I was wrong and it is actually standardized:


https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html#:~:text=%24XDG_CONFIG_HOME%20defines%20the%20base%20directory,state%20files%20should%20be%20stored.
> If, when attempting to write a file, the destination directory is non-existent an attempt should be made to create it with permission 0700. If the destination directory exists already the permissions should not be changed.

**Review by polybassa — APPROVED** [review]

**Pull request merged**